lib/ui/painting.dart

```diff
@@ -136,10 +136,10 @@ class Color {
   ///
   /// See also [fromARGB], which takes the opacity as an integer value.
   const Color.fromRGBO(int r, int g, int b, double opacity) :
-    value = ((((opacity * 0xff ~/ 1) & 0xff) << 24) |
-              ((r                    & 0xff) << 16) |
-              ((g                    & 0xff) << 8)  |
-              ((b                    & 0xff) << 0)) & 0xFFFFFFFF;
+    value = (((((opacity * 0xff + 0.5) ~/ 1) & 0xff) << 24) | // Round the alpha value for accuracy
```

Comment:
How about just `((opacity * 0xff).round() & 0xff) < 24` ? Seems like that would be more idiomatic.

Reply:
It is a constant which is why I think it was that way originally

Reply:
Ah, yeah, makes sense.

Reply:
can you add a comment explaining that? thanks